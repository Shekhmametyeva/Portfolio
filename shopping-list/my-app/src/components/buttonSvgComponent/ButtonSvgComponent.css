.button {
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0 10px;
}
.icon {
    width: 30px;
    height: 45px;
    fill: #2b2b2b;
}
.shopping__icon {
    width: 25px;
    height: 25px;
    fill: #2b2b2b;
}
.button:hover .icon{
    fill: #05dcb4;
}
.close {
    transform: rotate(45deg);
}
