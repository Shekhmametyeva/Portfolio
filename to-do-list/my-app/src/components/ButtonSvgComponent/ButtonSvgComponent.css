.button {
    border: none;
    width: 40px;
    height: 40px;
    cursor: pointer;
    background-color: transparent;
}
.icon {
    width: 100%;
    height: 100%;
}
.button:hover svg {
    fill: #016801;
}
.todo__button {
    width: 30px;
    height: 30px;
}
.todo__button:hover {
    border-color: #016801;
}
