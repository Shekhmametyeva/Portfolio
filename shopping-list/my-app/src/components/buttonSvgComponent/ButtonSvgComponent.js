import React from 'react';
import './ButtonSvgComponent.css';
import IconsSVG from '../../assets/icons/sprite.svg';

class ButtonSvgComponents extends React.Component {

    isCallback() {
        if(this.props.callback) {
            this.props.callback(this.props.value)
            
        }
        return
    }
    
    render() {
        const isClass = this.props.class ? `icon ${this.props.class}` : `icon`;
        
        return(
            <button type={this.props.type} className='button' onClick={() => this.isCallback() }>
                <svg className={isClass}>
                    <use xlinkHref={`${IconsSVG}#${this.props.name}`} />
                </svg>
            </button>
        )
    }
}

export default ButtonSvgComponents;