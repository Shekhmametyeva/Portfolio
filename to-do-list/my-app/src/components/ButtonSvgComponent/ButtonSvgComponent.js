import './ButtonSvgComponent.css'
import iconsSVG from '../../assets/icons/sprite.svg';


function ButtonSvgComponent (props) {
    return (
        <button 
            className={props.class ? `button ${props.class}` : `button`} 
            type={props.type} 
            onClick={() => {
                if(props.callback) {
                    props.callback();
                }
                
            }}>
                <svg className={props.class ? 'todo__icon' : 'icon'}>
                    <use xlinkHref={`${iconsSVG}#${props.name}`} />
                </svg>
        </button>
    )
}

export default ButtonSvgComponent