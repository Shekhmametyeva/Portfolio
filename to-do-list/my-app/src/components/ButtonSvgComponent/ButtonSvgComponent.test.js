import {render, screen} from '@testing-library/react';
import ButtonSvgComponent from './ButtonSvgComponent';

describe('тестируем ButtonSvgComponent', () => {
    test('render ButtonSvgComponent', () => {
        render(<ButtonSvgComponent name='add'/>);
        const btn = screen.getByRole('button');
        expect(btn).toBeInTheDocument();
        expect(btn).toMatchSnapshot();
    });
    test('render ButtonSvgComponent disabled', () => {
        render(<ButtonSvgComponent disabled={true} name='add'/>);
        const btn = screen.getByRole('button');
        expect(btn).toBeDisabled();
    });
    test('render ButtonSvgComponent active', () => {
        render(<ButtonSvgComponent disabled={false} name='menu'/>);
        const btn = screen.getByRole('button');
        expect(btn).toBeEnabled()
    });
    test('render ButtonSvgComponent with className', () => {
        render(<ButtonSvgComponent name='add' class='todo__button todo__done'/>);
        const btn = screen.getByRole('button');
        expect(btn).toHaveClass('todo__button todo__done');
    });
})