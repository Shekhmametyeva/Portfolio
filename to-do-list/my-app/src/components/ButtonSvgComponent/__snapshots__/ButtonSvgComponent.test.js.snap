// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестируем ButtonSvgComponent render ButtonSvgComponent 1`] = `
<button
  class="button"
>
  <svg
    class="icon"
  >
    <use
      xlink:href="sprite.svg#add"
    />
  </svg>
</button>
`;


